shiny/experimental/ui/_valuebox.py

```diff
@@ -5,29 +5,25 @@
 
 from htmltools import Tag, TagAttrs, TagAttrValue, TagChild, css, div
 
-from ..._typing_extensions import TypeGuard
 from ._card import card, card_body
 from ._card_item import CardItem
 from ._css import CssUnit, validate_css_unit
-from ._fill import bind_fill_role
+from ._fill import as_fill_carrier
 from ._layout import layout_column_wrap
-
-
-def is_01_scalar(x: object) -> TypeGuard[float]:
-    return isinstance(x, float) and x >= 0.0 and x <= 1.0
+from ._utils import consolidate_attrs, is_01_scalar
 
 
 # It seems to be to use % over fr here since there is no gap on the grid
-def to_width_unit(x: str | float) -> str:
-    if isinstance(x, float):
+def to_width_unit(x: str | float | int | numbers.Number) -> str:
+    if isinstance(x, numbers.Number):
```

Comment:
`numbers.Number` will match for weird things like `complex(2, 0)`, and `Fraction(2, 1)`, when we really just want floats and ints.

```suggestion
def to_width_unit(x: str | float | int) -> str:
    if isinstance(x, (float, int)):
```